h2o-r/h2o-package/R/mli.R

```diff
@@ -0,0 +1,2209 @@
+      geom_point_or_line(!is.numeric(newdata[[column]]),
+                         if (is.factor(pdp[[col_name]])) {
+                           ggplot2::aes(shape = "Partial Dependence", group = "Partial Dependence")
+                         } else {
+                           ggplot2::aes(linetype = "Partial Dependence", group = "Partial Dependence")
+                         },
+                         data = pdp, color = "black"
+      ) +
+      ggplot2::geom_rug(ggplot2::aes(x = .data[[col_name]], y = NULL, text = NULL),
+                        sides = "b", alpha = 0.1, color = "black",
+                        data = stats::setNames(as.data.frame(newdata[[column]]), col_name)
+      ) +
+      ggplot2::labs(y = "Response", title = sprintf(
+        "Individual Conditional Expectations on \"%s\"%s\nfor Model: \"%s\"", col_name,
+        if (is.null(target)) {
+          ""
+        } else {
+          sprintf(" with Target = \"%s\"", target)
+        },
+        model@model_id
+      )) +
+      ggplot2::scale_color_viridis_d(option = "plasma") +
+      ggplot2::scale_linetype_manual(values = c("Partial Dependence" = "dashed")) +
+      # make the histogram closer to the axis. (0.05 is the default value)
+      ggplot2::scale_y_continuous(expand = ggplot2::expansion(mult = c(0, 0.05))) +
+      ggplot2::theme_bw() +
+      ggplot2::theme(
+        legend.title = ggplot2::element_blank(),
+        axis.text.x = ggplot2::element_text(angle = if (h2o.isfactor(newdata[[col_name]])) 45 else 0, hjust = 1),
+        plot.margin = margin
+      )
+    return(p)
+  })
+}
+
+######################################## Explain ###################################################
+
+#' Generate explanations for \code{object} on \code{test_frame}.
+#'
+#' @param object One of the following: an H2OAutoML, an H2OAutoML Leaderboard slice, a model, a list of models.
+#' @param test_frame An H2OFrame.
+#' @param columns_of_interrest A vector of feature names. If specified, plot just these columns.
+#' @param include_explanations If specified, do only the specified explanations.
+#'   (Mutually exclusive with exclude_explanations)
+#' @param exclude_explanations Exclude specified explanations.
+#' @param top_n_features If \code{columns_of_interrest} is missing, use top n features.
+#' @param best_of_family If True, explain only best of family models
+#' @param user_overrides Overrides for individual explanations, e.g.,
+#'   list(shap_summary_plot = list(top_n_features = 50))
+#'
+#' @return List of outputs with class "explanation"
+#' @export
+h2o.explain <- function(object,
+                        test_frame,
+                        columns_of_interrest = NULL,
+                        include_explanations = "ALL",
+                        exclude_explanations = character(),
+                        top_n_features = 5,
+                        best_of_family = TRUE,
+                        user_overrides = list()) {
```

Comment:
If this is just for plot parameter overrides, then maybe we should use a parameter name like `plot_overrides` or `plot_arguments`?

Reply:
will change to `plot_overrides`